adapter_nrf528xx.go

```diff
@@ -219,6 +219,15 @@ func handleEvent() {
 					}
 				}
 			}
+		case C.BLE_GATTC_EVT_READ_RSP:
+			readEvent := gattcEvent.params.unionfield_read_rsp()
+			if debug {
+				println("evt: read response, data length", readEvent.len)
+			}
+			readingCharacteristic.handle_value.Set(readEvent.handle)
+			readingCharacteristic.offset = readEvent.offset
+			// Create a Go slice from the data.
+			readingCharacteristic.value = (*[255]byte)(unsafe.Pointer(&readEvent.data[0]))[:readEvent.len:readEvent.len]
```

Comment:
I do not think it is safe to retain the buffer after the event. The event buffer might have been reused for something else after the interrupt is handled. For example multiple events might occur at the same time.

A solution to this would be to provide a buffer to fill in `readingCharacteristic` and copy the data (and length) to it.

Reply:
That is why I have this code: https://github.com/tinygo-org/bluetooth/blob/read-characteristics-improved/gattc_sd.go#L440-L442

Reply:
No, that's not enough. Multiple events can come in before the buffer is copied, thus there is a race condition between copying the value and the next event interrupt arriving.

Reply:
OK, changes made to accommodate this.